src/extensionConsole/pages/mods/ModsPage.test.tsx

```diff
@@ -0,0 +1,290 @@
+    );
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("An error occurred")).toBeInTheDocument();
+    });
+    expect(screen.getByText("Failed to load mods")).toBeInTheDocument();
+  });
+
+  it("renders error state when marketplace listings fail to load", async () => {
+    appApiMock.onGet(API_PATHS.MARKETPLACE_LISTINGS).reply(500);
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("An error occurred")).toBeInTheDocument();
+    });
+    expect(screen.getByText("Internal Server Error")).toBeInTheDocument();
+  });
+
+  it("renders GetStartedView when tab key is active", async () => {
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+          dispatch(
+            modsPageSlice.actions.setActiveTab(MODS_PAGE_TABS.getStarted),
+          );
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.getByText("Welcome to the PixieBrix Extension Console"),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test("renders active mods when there are mods", async () => {
+    const mod1 = modDefinitionFactory({
+      metadata: modMetadataFactory({ name: "Test Mod 1" }),
+    });
+    const mod2 = modDefinitionFactory({
+      metadata: modMetadataFactory({ name: "Test Mod 2" }),
+    });
+    const mod3 = modDefinitionFactory({
+      metadata: modMetadataFactory({ name: "Test Mod 3" }),
+    });
+
+    useAllModDefinitionsMock.mockReturnValue(
+      valueToAsyncCacheState([mod1, mod2, mod3]),
+    );
+
+    const unavailableMod = modDefinitionFactory({
+      metadata: modMetadataFactory({ name: "Unavailable Test Mod" }),
+    });
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(
+            modComponentActions.activateMod({
+              modDefinition: mod1,
+              screen: "extensionConsole",
+              isReactivate: false,
+            }),
+          );
+          dispatch(
+            modComponentActions.activateMod({
+              modDefinition: unavailableMod,
+              screen: "extensionConsole",
+              isReactivate: false,
+            }),
+          );
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    waitForAct();
+    expect(screen.getByText("Test Mod 1")).toBeInTheDocument();
```

Comment:
`waitForAct` is unnecessary
```suggestion
    expect(screen.getByText("Test Mod 1")).toBeInTheDocument();
```